android/engine/src/main/java/org/smartregister/fhircore/engine/rulesengine/RulesFactory.kt

```diff
@@ -337,12 +337,21 @@ constructor(
     fun formatDate(date: Date, expectedFormat: String = "E, MMM dd yyyy"): String {
       return date.formatDate(expectedFormat)
     }
+
+    fun generateRandomIntLengthSix(): Int {
+      val uniqueId =
+        System.currentTimeMillis().toString().takeLast(6).toInt() +
+          (100000..maxInger).shuffled().random()
+      return if (uniqueId.toString().length > 6) uniqueId.toString().takeLast(6).toInt()
+      else uniqueId
+    }
   }
 
   companion object {
     private const val FHIR_PATH = "fhirPath"
     private const val DATA = "data"
     private const val TRUE = "true"
     private const val SERVICE = "service"
+    private const val maxInger = 999999
```

Comment:
```suggestion
    private const val inclusiveSixDigitMinimum = 100000
    private const val exclusiveSixDigitMaximum = 1000000
```

Reply:
Conventionally Kotlin constants should be  in [UPPER_SNAKE_CASE](https://developer.android.com/kotlin/style-guide#constant_names)

Reply:
cool, updated